soc/arm/st_stm32/common/stm32_backup_sram.c

```diff
@@ -23,10 +23,9 @@ static int stm32_backup_sram_init(const struct device *dev)
 	const struct stm32_backup_sram_config *config = dev->config;
 
 	int ret;
-	const struct device *clk;
 
-	/* enable backup SRAM clock */
-	clk = device_get_binding(STM32_CLOCK_CONTROL_NAME);
+	/* enable clock for subsystem */
+	const struct device *clk = DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE);
 	__ASSERT_NO_MSG(clk);
```

Comment:
I guess assert can also be removed now, right?

Reply:
Yes, the assert should be removed.

Reply:
ok, txs removed